### PR TITLE
NEW: Deprecated PDO in favour of native drivers

### PR DESCRIPTION
**Pull request by sminnee**
Addresses https://github.com/silverstripe/silverstripe-framework/issues/8598

Related: https://github.com/silverstripe/silverstripe-installer/pull/256

### CODE REVIEW

**Comment by sminnee** [comment]
For clarity: this doesn't change anything if SS_DATABASE_CLASS is not defined. That was already defaulting to non-PDO MySQL. It makes the following changes:

 * If you use the UI installer, it will default to non-PDO
 * If you use PDO and your Deprecation::notification_version is set to 4.5 or higher, then you will get a deprecation notice.

Note that by default notification_version is set to 4.0, presumably so that minor release upgrades don't start throwing notice level errors without an explicit chagne.

**Comment by robbieaverill** [comment]
Marked as accepted now

File: _register_database.php [hunk 1]
```diff
@@ -37,3 +19,20 @@
 			PHP extension is not available. Please install or enable it and refresh this page.'
     )
 );
+
+// Register MySQL PDO as a database adapter (listed as first option in Dev/Install/config-form.html)
```
Comment (robbieaverill) [inline _register_database.php]: I _think_ that moving this code down will make the last part of this sentence invalid. Could you remove it?

Comment (robbieaverill) [reply _register_database.php]: ping @sminnee

Comment (sminnee) [reply _register_database.php]: fixed

**Pull request merged**